src/platform/forms/save-in-progress/RoutedSavableApp.jsx

```diff
@@ -112,7 +114,20 @@ class RoutedSavableApp extends React.Component {
       newProps.prefillStatus !== this.props.prefillStatus &&
       newProps.prefillStatus === PREFILL_STATUSES.unfilled
     ) {
-      newProps.router.push(this.getFirstNonIntroPagePath(newProps));
+      let newRoute;
+      const { formConfig = {} } = newProps;
+      const { saveInProgress = {} } = formConfig;
+      if (
+        newProps.isStartingOver &&
+        typeof saveInProgress.restartFormCallback === 'function' &&
+        restartShouldRedirect(formConfig.wizardStorageKey || 'formStatus')
+      ) {
+        // Restart callback can optionally return a new route
+        newRoute = saveInProgress?.restartFormCallback() || null;
```

Comment:
Blech. Optionally return a new route? I'd prefer to see a consistent return type to reduce our Surface Area for Weirdness. (A phrase which I may hereafter officially coin! :sweat_smile:)

What are the use cases for returning a new route vs not returning a new route?

Reply:
Originally when I made these changes, the `restartFormCallback` was called more than once. Now it should only get called once, so yeah, making it always return a route is a good idea.

Reply:
I updated the comment and removed the `|| null`, but the actual `router.push` still has a fallback to the first page of the form. Should it not do that?

Reply:
Also, if you want... I've included the docs for this change here: https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/pull/586